src/main/java/com/rtb/projectmanagementtool/private-comments/PrivateCommentListServlet.java

```diff
@@ -0,0 +1,102 @@
+/** Servlet responsible for privatecomments */
+package com.rtb.projectmanagementtool.privatecomment;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.rtb.projectmanagementtool.auth.*;
+import com.rtb.projectmanagementtool.task.*;
+import com.rtb.projectmanagementtool.user.*;
+import java.io.IOException;
+import java.util.*;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/private-comment")
+public class PrivateCommentListServlet extends HttpServlet {
+  DatastoreService datastore;
+  AuthOps auth;
+
+  public PrivateCommentListServlet() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    auth = new AuthOps(datastore);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    Long userID = Long.parseLong(request.getParameter("userID"));
+    Long currentUser = auth.whichUserIsLoggedIn(request, response);
+
+    // controllers
+    UserController userController = new UserController(datastore);
+    TaskController taskController = new TaskController(datastore);
+    PrivateCommentController privateCommentController = new PrivateCommentController(datastore);
+
+    // get user
+    UserData user = userController.getUserByID(userID);
+
+    // get this user's tasks
+    ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
+
+    // init array of color task will appear
+    String[] colors = new String[tasks.size()];
+
+    // build list of private comments and colors
+    ArrayList<PrivateCommentData> privateComments = new ArrayList<>();
+
+    // only fill if the user is viewing their own profile
+    if (userID == currentUser) {
+      for (int i = 0; i < tasks.size(); i++) {
+        Long taskID = tasks.get(i).getTaskID();
+        PrivateCommentData privateComment =
```

Comment:
This would do multiple datastore queries to load all private comments for a single user.  Queries can get expensive because of the back and forth between the server and the database.  Anyway to make this more efficient?

Reply:
I'm not sure. I like the loop here because it gets each comment in the order that the tasks will be loaded as well. Would you suggest modifying the .getPrivateCommentByTaskID method? If so, I'm not exactly sure what would change.

Reply:
One thought would be to load all private comments owned by a single user.  If the assumption is these will only be viewable by the current user in their own user profile pages, this would mean loading all for the user would give you all the records you'll need.